Client/Application/LeanplumIntegration.swift

```diff
@@ -245,4 +267,78 @@ class LeanplumIntegration {
         }
         return LeanplumSettings(environment: environment, appId: appId, key: key)
     }
+    
+    // This must be called before `Leanplum.start` in order to correctly setup
+    // custom message templates.
+    private func setupCustomTemplates() {
+        // These properties are exposed through the Leanplum web interface.
+        // Ref: https://github.com/Leanplum/Leanplum-iOS-Samples/blob/master/iOS_customMessageTemplates/iOS_customMessageTemplates/LPMessageTemplates.m
+        let args: [LPActionArg] = [
+            LPActionArg(named: LPMT_ARG_TITLE_TEXT, with: LPMT_DEFAULT_ASKT_TO_ASK_TITLE),
+            LPActionArg(named: LPMT_ARG_TITLE_COLOR, with: UIColor.black),
+            LPActionArg(named: LPMT_ARG_MESSAGE_TEXT, with: LPMT_DEFAULT_ASK_TO_ASK_MESSAGE),
+            LPActionArg(named: LPMT_ARG_MESSAGE_COLOR, with: UIColor.black),
+            LPActionArg(named: LPMT_ARG_ACCEPT_BUTTON_TEXT, with: LPMT_DEFAULT_OK_BUTTON_TEXT),
+            LPActionArg(named: LPMT_ARG_CANCEL_ACTION, withAction: nil),
+            LPActionArg(named: LPMT_ARG_CANCEL_BUTTON_TEXT, with: LPMT_DEFAULT_LATER_BUTTON_TEXT),
+            LPActionArg(named: LPMT_ARG_CANCEL_BUTTON_TEXT_COLOR, with: UIColor.gray)
+        ]
+        
+        let responder: LeanplumActionBlock = { (context) -> Bool in
+            guard let context = context else {
+                return false
+            }
+            
+            // Don't display permission screen if they have already allowed/disabled push permissions
+            if self.profile?.prefs.boolForKey(applicationDidRequestUserNotificationPermissionPrefKey) ?? false {
+                FxALoginHelper.sharedInstance.readyForSyncing()
+                return false
+            }
+            
+            // Present Alert View onto the current top view controller
+            let rootViewController = UIApplication.topViewController()
+            let title = NSLocalizedString(context.stringNamed(LPMT_ARG_TITLE_TEXT), comment: "")
+            let message = NSLocalizedString(context.stringNamed(LPMT_ARG_MESSAGE_TEXT), comment: "")
+            let alert = UIAlertController(title: title, message: message, preferredStyle: UIAlertControllerStyle.alert)
+            
+            let cancelText = NSLocalizedString(context.stringNamed(LPMT_ARG_CANCEL_BUTTON_TEXT), comment: "")
+            alert.addAction(UIAlertAction(title: cancelText, style: UIAlertActionStyle.cancel, handler: { (action) -> Void in
+                // Log cancel event and call ready for syncing
+                context.runTrackedActionNamed(LPMT_ARG_CANCEL_ACTION)
+                FxALoginHelper.sharedInstance.readyForSyncing()
+            }))
+            
+            let acceptText = NSLocalizedString(context.stringNamed(LPMT_ARG_ACCEPT_BUTTON_TEXT), comment: "")
+            alert.addAction(UIAlertAction(title: acceptText, style: UIAlertActionStyle.default, handler: { (action) -> Void in
```

Comment:
Same re: `NSLocalizedString` and omitting the class name when the compiler can infer types.